### ISSUE TITLE
Would anyone be interested in "Drives", "Recent Files" and "Desktop Folder" dynamic submenus?

### ISSUE THREAD

**Issue by cheapy** [bbidulock/icewm#372]
I created a menuprog dynamic submenu running a script that lists the block devices on the fly and then parses the list and translates it into a menu file where it generates a menu option for device to open it.

If anyone is interested, I will post the code.  I wrote it for antiX 19 but could easily adjust the code generated for any of the IceWM based setups I have seen.

PS: as you can see from the test it allows for hard drives, external USB hard drives and flash drives, CDROM/DVD and Floppy as well as SDHC cards

I have added a "Recent Files" option and also a "Desktop Folder" option and am hoping they will be included in antiX19.  For other distros, obviously the menu and preferences are just showing how I would like to implement, and the icons or icon paths will need to be adjusted, as well as the file manager being used.  The code hasn't been heavily tested at this point, so if you find any issues, please let me know.

PSS: The Desktop Folder option is also recursive and creates a branch of a submenu for each folder within a folder so that someone could have folders in their Desktop that each contained items and/or other folders

![icewm-menu5-desktop-folder_1](https://user-images.githubusercontent.com/7473967/65293908-6b796680-db22-11e9-8752-18541b9cd7df.jpg)

![antiX 19 Cobalt-Blue with Drives menu](https://user-images.githubusercontent.com/7473967/62618141-1b7b7300-b8d9-11e9-81c9-445ff0ff4d98.jpg)

**Comment by bbidulock** [comment]
Wow.  Looks like one of my old drives (full of Linux distro partitions).  Sure, post up the code: maybe we can add it to contrib or something.

**Comment by cheapy** [comment]
Note that I added more features and uploaded the code.  New version 6 uploaded today.

**Comment by gijsbers** [comment]
Where is it uploaded? Could you give a URL where we can download it from?

**Comment by cheapy** [comment]
Up Above

(new version 7 posted below)

If you find any troubles or think of something it needs, let me know :)

**Comment by cheapy** [comment]
And BTW, I forgot to say "THANKS!" for adding the menuprog feature.  That was a great innovation.

**Comment by cheapy** [comment]
I've now added the ability of my Recent Files option to list many files by organizing them in lists of 15 per menu with a "Next 15" option at the bottom of each.  I set it for 100 files as a guess or what delay would be acceptable for it to build the list and menus taking 4 seconds on a fast machine.  I did this because whenever I go looking for something "Recent", the list always seems too short because whatever I am looking for almost always has already scrolled out of the top 10 or 20.  It could be made configurable I suppose, but I'm not sure where that should be stored or how normal folks would maintain it.  I will upload new code after I finish testing it a bit more.  I should try it on my antique Pentium III, as IceWM seems to be used on old machines a lot due to its great performance even on old, low resource machines.

**Comment by cheapy** [comment]
Here is the newest version where it loads the 70 most recent files and displays them in groups of 10.

new improved code and zip file below

**Comment by cheapy** [comment]
If anyone tries these, feedback would be appreciated, be it good or with bug reports so I can hunt them down.  At home here, I have reduced it to loading 50 on the recent files option to keep the delay to a minimum.

**Comment by cheapy** [comment]
New code sets a default of displaying 10 recent files, but adds an option to change that setting, and in addition an option to clear the history.  Also moved the building of the menu off the main menu so that it doesn't try to build it while you accidentally hover over it while navigating the main menu.  I'm running at 50 on my faster machine and set it for 20 on the old Pentium III.

[icewm-menu-9.zip](https://github.com/bbidulock/icewm/files/3650230/icewm-menu-9.zip)

PS: this version requires YAD to run the set default and clear history scripts.
PSS: found that my new default setting routine was failing, fixed it, and now at zip file version 9

![icewm-menu-8-recentfiles0](https://user-images.githubusercontent.com/7473967/65478594-ec509f00-de4f-11e9-9712-37f48efefa46.jpg)

**Comment by cheapy** [comment]
Can I ask if there is a way to "localize" the text shown for entries keyed in on a menu, especially menuprog and menufile entries?

**Comment by gijsbers** [comment]
Not within icewm. The file which menufile loads or the output of menuprog should be localized.

**Comment by cheapy** [comment]
[icewm-menu-12.zip](https://github.com/bbidulock/icewm/files/3722574/icewm-menu-12.zip)

In my version 12 I added localization for my 3 options and the YAD scripts provided by PPC, which I tested in Spanish.  I don't at this point know how localization is done for antiX's main menu (different icon sets * many languages).  My programs were designed to run as "menuprog" dynamic menu's, so that  does cause a maintenance issue for antiX, but it looks like it will be included in the English version anyway.
